mParticle-Appboy/MPKitAppboy.m

```diff
@@ -192,9 +191,15 @@ - (NSString *)advertisingIdentifierString {
 
 #pragma mark ABKAppboyEndpointDelegate
 - (NSString *)getApiEndpoint:(NSString *)appboyApiEndpoint {
-    NSString *host = self.host;
-    NSString *endpoint = [appboyApiEndpoint stringByReplacingOccurrencesOfString:originalHost withString:host];
-    return endpoint;
+    NSMutableString *modifiedEndpoint = [appboyApiEndpoint mutableCopy];
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:
+                                  @"https.*\\.com" options:0 error:nil];
```

Comment:
Since regex is greedy by default this would result in the longer of two matches if the query string happened to contain the string `.com`. My suggestion is to match something like `https?:\\/\\/[^/]*\\.com` and add a unit test for e.g. https://example.com/?foo=bar.com

Typically it's also possible to configure the regex to be non-greedy but I've found replacing `.*` with `[^<relevant char>]*` is usually a pretty good approach and helps make regex more robust.